java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -1364,6 +1369,18 @@ JNIEXPORT jint JNICALL Java_ai_rapids_cudf_ColumnVector_getNativeTypeId(JNIEnv *
   CATCH_STD(env, 0);
 }
 
+JNIEXPORT jint JNICALL Java_ai_rapids_cudf_ColumnVector_getNativeTypeScale(JNIEnv *env,
+                                                                        jobject j_object,
+                                                                        jlong handle) {
+  JNI_NULL_CHECK(env, handle, "native handle is null", 0);
+  try {
+    cudf::jni::auto_set_device(env);
+    cudf::column_view *column = reinterpret_cast<cudf::column_view *>(handle);
+    return static_cast<jint>(column->type().scale());
```

Comment:
is type checking needed before calling `scale()` on  type_id? will it blow up in case its not a decimal type?

Reply:
We should not need the cast and I would prefer to not have it because it shows that we don't have issues mapping from the native scale to the JNI representation.